tracer/src/Datadog.Trace.ClrProfiler.Native/pal.h

```diff
@@ -97,6 +98,40 @@ inline int GetPID()
 #endif
 }
 
+inline WSTRING GetCurrentModuleFileName()
+{
+    static WSTRING moduleFileName = EmptyWStr;
+    if (moduleFileName != EmptyWStr)
+    {
+        // use cached version
+        return moduleFileName;
+    }
+
+#ifdef _WIN32
+    HMODULE hModule;
+    if (GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS, (LPCTSTR)GetCurrentModuleFileName,
```

Comment:
Consider adding GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT so that the module handle doesn't have to be closed.

Reply:
good catch! will do.

Reply:
done in bf3ef8d